### PR TITLE
Cxf-soap: Extend test coverage with Converter scenario #4652

### PR DESCRIPTION
**Pull request by JiriOndrusek**
fixes https://github.com/apache/camel-quarkus/issues/4652

Adds CSB test ([PayLoadConvertToPOJOTest](https://github.com/apache/camel-spring-boot/blob/camel-spring-boot-3.x/components-starter/camel-cxf-soap-starter/src/test/java/org/apache/camel/component/cxf/soap/springboot/converter/PayLoadConvertToPOJOTest.java)) for camel-quarkus.

Adds CSB test ([CxfConsumerPayLoadConverterTest](https://github.com/apache/camel-spring-boot/blob/camel-spring-boot-3.x/components-starter/camel-cxf-soap-starter/src/test/java/org/apache/camel/component/cxf/soap/springboot/CxfConsumerPayLoadConverterTest.java)) is for camel-quarkus.

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Comment by aldettinger** [comment]
Maybe to be merged after the current pending release.

**Comment by JiriOndrusek** [comment]
> Maybe to be merged after the current pending release.

No problem, I'll switch PR to draft.

**Review by ppalaga — COMMENTED** [review]
We need to find a proper answer to the camel-quarkus-jaxb question. Otherwise very nice, thanks @JiriOndrusek !

File: integration-tests/cxf-soap-grouped/pom.xml [hunk 1]
```diff
@@ -72,6 +72,10 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-resteasy</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.apache.camel.quarkus</groupId>
+            <artifactId>camel-quarkus-jaxb</artifactId>
+        </dependency>
```
[the remark this answers is not in the document]
Comment (ffang) [reply integration-tests/cxf-soap-grouped/pom.xml:78]: Hi @JiriOndrusek  @ppalaga ,

The test counterpart in CSB depends on camel-jaxb, needs to use the camel-jaxb converter there.

Freeman

[the remark this answers is not in the document]
Comment (ppalaga) [reply integration-tests/cxf-soap-grouped/pom.xml:78]: OK, it sounds like the separate jaxb dependency is expected then. Let's keep it there.

**Comment by ppalaga** [comment]
It is OK to leave the cq-jaxb dependecy in the test. @JiriOndrusek could you please rebase?

**Comment by JiriOndrusek** [comment]
@ppalaga I just recalled, that this change can not be merged. There is a problem (whih happens on CQ not CSB with converters). @ffang is working on it. See https://issues.apache.org/jira/browse/CAMEL-19160

**Comment by JiriOndrusek** [comment]
@ppalaga This PR can be merged, to allow backporting.

**Review by ppalaga — COMMENTED** [review]
One small nitpick inline, otherwise 👍

**Review by ppalaga — COMMENTED** [review]
One small nitpick inline, otherwise 👍

**Comment by JiriOndrusek** [comment]
All fixed, can be merged.

**Comment by ppalaga** [comment]
Perfect, thank @JiriOndrusek

**Pull request merged**